design/ceph/multus-network.md

```diff
@@ -78,21 +78,71 @@ Nothing to do in particular since they don't use any service IPs.
+
+The Rook-Ceph Operator's CSI controller creates a DaemonSet configured to use the
+`network.selectors.public` network specified for the CephCluster. This DaemonSet runs on all the
+nodes that will have CSI pods. Its Pods exist to "hold" a particular network interface that the CSI
+pods can reliably connect to. These Pods will be referred to as the "holder" Pods.
+
+TODO: VERIFY THIS, MAYBE IT'S UNECCESSARY: These Pods should only be stopped and restarted when a
+node is stopped so that volume operations do not become blocked. The Rook-Ceph Operator's CSI
+controller should place a finalizer on the Pods so that they cannot be accidentally removed, and we
+should make note that even if the holder DaemonSet is updated, the Pods will only be changed on a
+node reboot.
+
+Furthermore, a new Rook-specific CSI sidecar will run in CSI plugin Pods alongside the Ceph CSI
+plugin process. This sidecar will watch for holder Pods, and when a holder Pod is created or
+updated, it will move the public network from the Pod's network namespace to the host node's network
+namespace. In this way, even if the CSI pods are restarted, the network interface will not change,
+and the operations to the Ceph volume will not be blocked.
+
+If a holder Pod is removed, the sidecar should remove the interface from the host's network
+namespace. Similarly, if the sidecar container is stopped, it should move all holder Pod namespaces
+from the host's network namespace back to the holder Pod's network namespace.
+
+The sidecar must:
+- be a privileged container
+- have SYS_ADMIN and NET_ADMIN capabilities
+- be on the host network namespace
+- have access to the /var/run/netns directory
+
+OLD: If an error occurs during the setup job, the job will clean up after itself, and the holder pod will
+restart. The holder pod will receive a fresh interface from multus, and try the migration again.
+NEW: If an error occurs in the sidecar during network migration, it will fail, causing the CSI
+plugin pod to restart and re-try migration until the operation succeeds.
+TODO: @leseb @travisn Does this seem an okay change?
+
+Restarting the node will cause the multus interface in the host namespace to go away. The holder pod
+will once again get an interface which will be migrated again via the sidecar.
+
+When a new node is added, a holder pod is added to it by the DaemonSet, and the migration process
+described above occurs on the node.
+
+The holder pod, the setup job, and the teardown job require a service account with the permissions
+to interact with pods and jobs. The rook-csi-multus service account was created to fulfill this need.
+
+The holder DaemonSet should be deleted when the CSI driver components are removed. The termination
+of the holder Pods triggers the sidecar to remove the multus interfaces from the host network
+namespaces of the nodes.
+
+The initial implementation of this design will be limited to supporting a single CephCluster with
```

Comment:
Agreed to just support a single cephcluster for now

Reply:
I'm not sure about the complexity of supporting multiple clusters. It's probably just looping over all the CephCluster with Multus annotations we find.